### ISSUE TITLE
Diagnostic Test Cartridge 2.0 - Different behaviour than real hardware

### ISSUE THREAD

**Issue by ale-79** [stella-emu/stella#189]
On my tests with real hardware, the rom (see issue #188) returns an error "6" (joysticks, according to the [atariprotos page](http://www.atariprotos.com/2600/software/diagnostic/diagnostic.htm)), without anything plugged in the controller ports.

Stella 5.0.1  returns  "8" (Collisions)

Other emulators behave differently, Mame seems to be correct here:
6502.ts        1 (Switches)
Javatari    1 (Switches)
Mame        6 (Joysticks)

**Comment by thrust26** [comment]
4.7.2 returns 6 too

**Comment by sa666666** [comment]
OK, so definitely a regression.

**Comment by sa666666** [comment]
I wonder, is this possibly related to #126?

**Comment by thrust26** [comment]
Nope, its collision related. CXBLPF to be precise and there the used bit 6 which is set when it shouldn't.

Didn't we have this before? This is the only unused collision bit 6, maybe the fix was only for bits 0..5?

**Comment by ale-79** [comment]
I think it was issue #173, but it was about write only TIA addresses, so didn't affect collision registers

**Comment by thrust26** [comment]
The fix is in TIA::peek():
```
    case CXBLPF:
      result = collCXBLPF();// | (lastDataBusValue & 0x40);
      break;
```
Probably a bit too much was fixed with #173. No Stella fails with 6 again. :)

**Comment by sa666666** [comment]
Looks like this bit of code was supposed to be removed??

**Comment by thrust26** [comment]
No, that's the fix. I wanted to show the removed code.

**Comment by sa666666** [comment]
Yes, I meant that the code you commented out should have been removed in a previous refactoring.  I notice that the ROM now takes a little longer to start up, matching the behaviour from Stella 4 in that regard too.  I will commit the fix in a moment.

**Comment by DirtyHairy** [comment]
Seems I am late to the party :) The commit is definitely correct; the previous behavior was an oversight that should have been fixed in #173
